File: src/main/kotlin/com/example/demo/model/StandingOrder.kt
package com.example.demo.model

import javax.persistence.*

@Entity
@Table(name = "StandingOrder")
data class StandingOrder(
    var quantity: Long = 0,
    var type: OrderType = OrderType.BUY,
    @Column(name = "limitValue") var limit: Long = 0,
    @ManyToOne @JoinColumn(name = "userId", nullable = false, referencedColumnName = "id") var user: User = User(),
    @GeneratedValue @Id val id: Long = -1
) {
    fun fill(volume: Long){
        quantity -= volume
    }
}

File: Models/StandingOrder.cs
namespace TestApi.Models
{
    public class StandingOrder{
        // public StandingOrder(decimal quantity, OrderType type, decimal limit, User user){
        //     this.quantity = quantity;
        //     this.type = type;
        //     this.limit = limit;
        //     this.user = user;
        // }
        public int Id { get; set; }

        public decimal quantity { get; set; } = 0;
        public OrderType type { get; set; } = OrderType.Buy;
        public decimal limit { get; set; } = 0;
        public User user = new User();


        public void fill(Decimal volume){
            this.quantity -= volume;
        }
    }
}